File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeCraftSlotCountIncreasedMessage.scala





















// Generated on 05/03/2013 10:34:53
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeCraftSlotCountIncreasedMessage extends MessageDeserializer {
  val messageId = 6125

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeCraftSlotCountIncreasedMessage.messageId

    val newMaxSlot: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(newMaxSlot)
    }

  }

  protected def create(newMaxSlot: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val newMaxSlot = buf.readByte()

    create(newMaxSlot)
  }

}

object ExchangeCraftSlotCountIncreasedMessage extends ExchangeCraftSlotCountIncreasedMessage {
  case class MessageImpl(newMaxSlot: Byte) extends Message
  type Target = MessageImpl

  override def create(newMaxSlot: Byte) = MessageImpl(newMaxSlot)

  def apply(newMaxSlot: Byte) = create(newMaxSlot)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeCraftSlotCountIncreasedMessage.java


// Generated on 05/08/2013 19:37:53
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeCraftSlotCountIncreasedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6125;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte newMaxSlot;
    
    public ExchangeCraftSlotCountIncreasedMessage() { }
    
    public ExchangeCraftSlotCountIncreasedMessage(byte newMaxSlot) {
        this.newMaxSlot = newMaxSlot;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(newMaxSlot);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        newMaxSlot = buf.readByte();
        if (newMaxSlot < 0)
            throw new RuntimeException("Forbidden value on newMaxSlot = " + newMaxSlot + ", it doesn't respect the following condition : newMaxSlot < 0");
    }
    
}
